cbmc/proofs/mqtt_state.c

```diff
@@ -285,41 +285,64 @@ bool valid_IotMqttNetworkInfo( const IotMqttNetworkInfo_t *pInfo )
  * IotMqttConnectInfo
  ****************************************************************/
 
+void bound_IotMqttConnectInfo( IotMqttConnectInfo_t *pInfo )
+{
+#ifdef SUBSCRIPTION_COUNT_MAX
+  __CPROVER_assume( pInfo->previousSubscriptionCount < SUBSCRIPTION_COUNT_MAX );
+#endif
+
+#ifdef CLIENT_IDENTIFIER_LENGTH_MAX
+  __CPROVER_assume( pInfo->clientIdentifierLength < CLIENT_IDENTIFIER_LENGTH_MAX &&
+                    VALID_CBMC_SIZE( pInfo->clientIdentifierLength ) );
+#else
+  __CPROVER_assume( pInfo->clientIdentifierLength <= UINT16_MAX );
+#endif
+
+#ifdef USER_NAME_LENGTH_MAX
+    __CPROVER_assume( pInfo->userNameLength < USER_NAME_LENGTH_MAX &&
+                      VALID_CBMC_SIZE( pInfo->userNameLength ) );
+#else
+  __CPROVER_assume( pInfo->userNameLength <= UINT16_MAX );
+#endif
+
+#ifdef PASSWORD_LENGTH_MAX
+    __CPROVER_assume( pInfo->passwordLength < PASSWORD_LENGTH_MAX &&
+                      VALID_CBMC_SIZE( pInfo->passwordLength ) );
+#else
+  __CPROVER_assume( pInfo->passwordLength <= UINT16_MAX );
+#endif
+}
+
 IotMqttConnectInfo_t *allocate_IotMqttConnectInfo( IotMqttConnectInfo_t *pInfo )
 {
   if ( pInfo == NULL ) pInfo = malloc_can_fail( sizeof( *pInfo ) );
   if ( pInfo == NULL ) return NULL;
 
+  bound_IotMqttConnectInfo( pInfo );
+
   pInfo->pPreviousSubscriptions =
     allocate_IotMqttSubscriptionArray( NULL,
 				       pInfo->previousSubscriptionCount );
+
   pInfo->pWillInfo = allocate_IotMqttPublishInfo( NULL );
   pInfo->pClientIdentifier = malloc_can_fail( pInfo->clientIdentifierLength );
   pInfo->pUserName = malloc_can_fail( pInfo->userNameLength );
   pInfo->pPassword = malloc_can_fail( pInfo->passwordLength );
+
+  return pInfo;
 }
 
 bool valid_IotMqttConnectInfo( const IotMqttConnectInfo_t *pInfo )
 {
   return
     pInfo != NULL &&
 
-    VALID_STRING( pInfo->pClientIdentifier, pInfo->clientIdentifierLength ) &&
-    VALID_CBMC_SIZE( pInfo->clientIdentifierLength ) &&
-
-    VALID_STRING( pInfo->pUserName, pInfo->userNameLength ) &&
-    VALID_CBMC_SIZE( pInfo->userNameLength ) &&
-
-    VALID_STRING( pInfo->pPassword, pInfo->passwordLength ) &&
-    VALID_CBMC_SIZE( pInfo->passwordLength ) &&
-
-    // MAX is one greater than the maximum length
-    pInfo->previousSubscriptionCount < SUBSCRIPTION_COUNT_MAX &&
+    valid_IotMqttPublishInfo( pInfo->pWillInfo ) &&
```

Comment:
I don't get it.  You deleted these properties here and just moved them to the Bounded function.

Reply:
See previous comments comments.

Reply:
OK